### PR TITLE
feat(ListChoice): add disabled boolean property

### PR DESCRIPTION
**Pull request by vepor**
Added `disabled` boolean property, so it can be used in the cases when it's pre-filled with `true` for `selected` property and it user should not be able to change the value. <br/><br/><br/><url>LiveURL: https://orbit-feat-listchoice-disabled.surge.sh</url>

### CODE REVIEW

File: packages/orbit-components/src/ListChoice/index.js [hunk 1]
```diff
@@ -41,19 +41,23 @@ const StyledListChoice = styled.div`
...
-      color: ${({ theme }) => theme.orbit.colorIconPrimary};
-    }
     outline: none;
+    ${({ disabled }) =>
+      !disabled &&
+      css`
+        background-color: ${({ theme }) => theme.orbit.paletteCloudLight};
+        ${StyledListChoiceIcon} svg {
+          color: ${({ theme }) => theme.orbit.colorIconPrimary};
+        }
```
Comment (mainframev) [inline packages/orbit-components/src/ListChoice/index.js:59]: destructure theme together with disabled :)

File: packages/orbit-components/src/ListChoice/index.js [hunk 2]
```diff
@@ -78,17 +82,19 @@ StyledListChoiceContent.defaultProps = {
...
     ...(selectable ? { "aria-checked": selected } : null),
   };
 
   return (
     <StyledListChoice
-      onClick={onClick}
+      onClick={!disabled ? onClick : null}
       data-test={dataTest}
-      onKeyDown={handleKeyDown(onClick)}
+      onKeyDown={!disabled ? handleKeyDown(onClick) : null}
```
Comment (mainframev) [inline packages/orbit-components/src/ListChoice/index.js:94]: i suppose here should be undefined, not null :)

File: packages/orbit-components/src/ListChoice/index.js.flow [hunk 3]
```diff
@@ -10,6 +10,7 @@ export type Props = {|
   +description?: Translation,
   +selectable?: boolean,
   +selected?: boolean,
+  +disabled?: boolean,
   +icon?: React$Node,
```
Comment (mainframev) [inline packages/orbit-components/src/ListChoice/index.js.flow:14]: i know, that you add only `disabled` prop, but here is `React$Node` below, you could replace it to `React.Node` too :)

File: packages/orbit-components/src/ListChoice/index.js (hunk 2 above)
Comment (vepor) [reply packages/orbit-components/src/ListChoice/index.js:94]: It's the same, it won't do anything 🤷

File: packages/orbit-components/src/ListChoice/index.js (hunk 1 above)
Comment (vepor) [reply packages/orbit-components/src/ListChoice/index.js:59]: Thanks, good catch! 🙂

File: packages/orbit-components/src/ListChoice/index.js.flow (hunk 3 above)
Comment (vepor) [reply packages/orbit-components/src/ListChoice/index.js.flow]: ✅

File: packages/orbit-components/src/ListChoice/index.js (hunk 2 above)
Comment (mainframev) [reply packages/orbit-components/src/ListChoice/index.js:94]: i know :) it's SC, in normal components we usually pass undefined in such cases, so i would follow same logic here, makes no difference in the end, but for consistency. Also i suppose it's better from perspective `null` and `undefined` differences, like here is more about something that hasn't been initialized

**Review by mainframev — APPROVED** [review]

File: packages/orbit-components/src/ListChoice/index.js (hunk 2 above)
Comment (vepor) [reply packages/orbit-components/src/ListChoice/index.js:94]: That would mean that we have to go through the whole code base, we're not consistent when comes to usage of `null` or `undefined`.

**Pull request merged**